docroot/modules/humsci/hs_actions/src/Plugin/Action/CloneNode.php

```diff
@@ -0,0 +1,176 @@
+    $values = range(1, 10);
+    $form['clone_count'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Clone how many times'),
+      '#options' => array_combine($values, $values),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['clone_count'] = $form_state->getValue('clone_count');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access($object, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    /** @var \Drupal\node\NodeInterface $object */
+    $result = $object->access('update', $account, TRUE)
+      ->andIf($object->getOwner()->access('edit', $account, TRUE));
+
+    return $return_as_object ? $result : $result->isAllowed();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    for ($i = 0; $i < $this->configuration['clone_count']; $i++) {
+      $duplicate_node = $this->duplicateEntity($entity);
+      $duplicate_node->save();
+    }
+  }
+
+  /**
+   * Recursively clone an entity and any dependent entities in reference fields.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   Entity to clone.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityInterface
+   *   Cloned entity.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function duplicateEntity(ContentEntityInterface $entity) {
+    $duplicate_entity = $entity->createDuplicate();
+
+    // Loop through paragraph and eck fields to clone those entities.
+    foreach ($this->getReferenceFields($entity->getEntityTypeId(), $entity->bundle()) as $field) {
+      foreach ($duplicate_entity->{$field->getName()} as $value) {
+        $value->entity = $this->duplicateEntity($value->entity);
+      }
+    }
+
+    return $duplicate_entity;
+  }
+
+  /**
+   * Get fields that need to have their referenced entities cloned.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID. Only entity types that implement.
+   * @param string $bundle
+   *   The bundle.
+   *
+   * @return \Drupal\field\Entity\FieldConfig[]
+   *   Array of fields that need cloned values.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function getReferenceFields($entity_type_id, $bundle) {
+    $fields = $this->entityFieldManager->getFieldDefinitions($entity_type_id, $bundle);
+
+    if ($this->entityTypeManager->hasDefinition('eck_entity_type')) {
+      $eck_types = $this->entityTypeManager->getStorage('eck_entity_type')
+        ->loadMultiple();
+      $clone_target_types = array_keys($eck_types);
+    }
+
+    $clone_target_types[] = 'paragraph';
+
+    // Filter out fields that we dont care about.
```

Comment:
What fields don't you care about and why? This could use a more detailed comment.